src/pages/workspaces/workspace/analysis/AzureComputeModal.js

```diff
@@ -0,0 +1,332 @@
+                max: 64000,
+                isClearable: false,
+                onlyInteger: true,
+                value: computeConfig.diskSize,
+                onChange: v => updateComputeConfig('diskSize', v)
+              })
+            ])
+          ])
+        ])
+      ])
+    ])
+  }
+
+  // Will be used once we support update
+  // const hasChanges = () => {
+  //   const existingConfig = adaptRuntimeDetailsToFormConfig()
+  //
+  //   return !_.isEqual(existingConfig, computeConfig)
+  // }
+  //
+  // const adaptRuntimeDetailsToFormConfig = () => {
+  //   return currentRuntimeDetails ? {
+  //     machineType: currentRuntimeDetails.runtimeConfig?.machineType || defaultAzureMachineType,
+  //     diskSize: currentRuntimeDetails.diskConfig?.size || defaultAzureDiskSize,
+  //     region: currentRuntimeDetails.runtimeConfig?.region || defaultAzureRegion
+  //   } : {}
+  // }
+
+  const doesRuntimeExist = () => !!currentRuntimeDetails
+
+  const renderActionButton = () => {
+    const commonButtonProps = {
+      tooltipSide: 'left',
+      disabled: Utils.cond([viewMode === 'deleteEnvironment', () => getIsRuntimeBusy(currentRuntimeDetails)],
+        () => doesRuntimeExist()),
+      tooltip: Utils.cond([viewMode === 'deleteEnvironment', () => getIsRuntimeBusy(currentRuntimeDetails) ? 'Cannot delete a runtime while it is busy' : undefined],
+        [doesRuntimeExist(), () => 'Update not supported for azure runtimes'],
+        () => undefined)
+    }
+
+    return h(ButtonPrimary, {
+      ...commonButtonProps,
+      onClick: () => applyChanges()
+    }, [
+      Utils.cond(
+        [viewMode === 'deleteEnvironment', () => 'Delete'],
+        [doesRuntimeExist(), () => 'Update'],
+        () => 'Create'
+      )
+    ]
+    )
+  }
+
+  // Helper functions -- begin
+  const applyChanges = _.flow(
+    Utils.withBusyState(setLoading),
+    withErrorReportingInModal('Error modifying cloud environment', onDismiss)
+  )(async () => {
+    //TODO: metrics onclick
+    //sendCloudEnvironmentMetrics()
+
+    //each branch of the cond should return a promise
+    await Utils.cond([viewMode === 'deleteEnvironment', async () => await Ajax().Runtimes.runtimeV2(workspaceId, currentRuntimeDetails.runtimeName).delete()], //delete runtime
+      [Utils.DEFAULT, async () => {
+        const disk = {
+          size: computeConfig.diskSize,
+          //We do not currently support re-attaching azure disks
+          name: Utils.generatePersistentDiskName(),
+          labels: { saturnWorkspaceNamespace: namespace, saturnWorkspaceName: workspaceName }
+        }
+
+        await Ajax().Runtimes.runtimeV2(workspaceId, Utils.generateRuntimeName()).create({
+          region: computeConfig.region,
+          machineSize: computeConfig.machineType,
+          labels: {
+            saturnWorkspaceNamespace: namespace,
+            saturnWorkspaceName: workspaceName
+          },
+          disk
+        })
+      }]
+    )
+
+    onSuccess()
+  })
```

Comment:
The section above could use a little care and feeding. I threw some time in your calendar to talk through it and hopefully finish things up.

Reply:
Yeah, there's some history involved here and some lessons learned that went into `renderActionButton` and `applyChanges`. Would be happy to talk about the approach, since a lot of future-proofing went into the thought process behind the respective implementations.